src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs

```diff
@@ -2,55 +2,193 @@
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
 using System;
+using System.Collections.Generic;
 using System.Threading;
 using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Logging;
 using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.Hosting;
 using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
 
 namespace Microsoft.Azure.WebJobs.Script.WebHost
 {
-    public class WebJobsScriptHostService : IHostedService, IDisposable
+    public class WebJobsScriptHostService : IHostedService, IScriptHostManager, IDisposable
```

Comment:
So this hosted service replaces ScriptHostManager "keep alive" functionality? Layering wise, how is a non WebHost ScriptHost spun up? Perhaps the below isn't WebHost specific and can be moved to WebJobs.Script?